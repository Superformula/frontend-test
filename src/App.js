import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import Header from 'components/Header';
import List from 'routes/List';
import Detail from 'routes/Detail';
import { withContextState, connectState } from 'decorators/contextState';
import 'reset-css';
import './base.css';


@withContextState({
  businesses: [],
  current: {},
  params: {
    location: 'Las Vegas',
    limit: 50
  }
})
class App extends Component {
  render() {
    return (
      <Router>
        <div>
          <Route exact path="/" component={List}  />
          <Route path="/:businessId" component={Detail} />
        </div>
      </Router>
    )
  }
}

export default App;
