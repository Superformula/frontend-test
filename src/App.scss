.App {
    background-color: white;
    color: black;
    height: 100vh;
    width: 100%;
    box-sizing: border-box;
    padding-left: 35px;
    padding-right: 35px; 
}