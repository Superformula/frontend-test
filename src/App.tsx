import React from 'react';
import './styles/core.scss';
import { BrowserRouter, Switch, Route, Redirect } from 'react-router-dom';
import { ApolloProvider } from '@apollo/react-hooks';

import { Restaurants } from 'screens/Restaurants/Restaurants';
import { RestaurantDetails } from 'screens/RestaurantDetails/RestaurantDetails';
import { client } from './api/client';

export const App: React.FC = () => (
  <ApolloProvider client={client}>
    <BrowserRouter>
      <Switch>
        <Route path="/" exact>
          <Restaurants />
        </Route>
        <Route path="/restaurants/:restaurantId">
          <RestaurantDetails />
        </Route>
        <Redirect to="/" />
      </Switch>
    </BrowserRouter>
  </ApolloProvider>
);
