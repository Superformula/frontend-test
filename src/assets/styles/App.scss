@import "variables";

@import "~reset-css/sass/reset";
@import "~include-media/dist/include-media";

@import "typography";
@import "mixins";
@import "layout";
