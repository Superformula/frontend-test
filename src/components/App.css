:root {
  --white: #fff;
  --light-gray: #e6e6e6;
  --gray: #666666;
  --dark-gray: #333333;
  --black: #000;
  --blue: #002b56;
  --border: #c8c8c8;
  --green: #00e8a4;
  --red: #ff3548;
}

:root {
  --space-sm: 4px;
  --space-md: 8px;
  --space-lg: 16px;
  --space-xl: 24px;
  --space-2xl: 32px;
}

* {
  font-family: HelveticaNeue;
}
