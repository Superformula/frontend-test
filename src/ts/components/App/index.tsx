// Imports
// -------

// Libraries
import React from 'react';

// Internal
// --------

const App: React.FC = () => {
    //
    // Render

    return (
        <div>
            Hello World
        </div>
    );
}

// Exports
// -------

export default App;
