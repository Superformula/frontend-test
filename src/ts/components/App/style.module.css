.app {
    margin: 36px 64px 0;
}
