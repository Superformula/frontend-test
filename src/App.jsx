import React from 'react';
import "./App.scss";

const App = () => <div className="App">App is Running!</div>;

export default App;