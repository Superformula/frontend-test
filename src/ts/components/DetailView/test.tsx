// Imports
// -------

// Libraries
import React from 'react';
import renderer from 'react-test-renderer';
// Components
import App from './';

// Internal
// --------

const getComponent = (props = {}) => renderer.create(
    <App {...props} />,
);

// Tests
// -----

describe('src/js/App', () => {
    let component;

    describe('Rendering', () => {
        it('renders default correctly', () => {
            component = getComponent();
            expect(component).toMatchSnapshot();
        });
    });
});
