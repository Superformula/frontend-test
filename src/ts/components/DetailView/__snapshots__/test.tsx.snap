// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/ts/components/DetailView Rendering renders default correctly 1`] = `
<div>
  <Hero
    title="Restaurants"
  />
</div>
`;
